homeassistant/components/light/lw12wifi.py

```diff
@@ -0,0 +1,200 @@
+    # Add devices
+    add_devices([LW12WiFi(name, host, port, rgb, brightness, effect,
+                          transition)])
+
+
+class LW12WiFi(Light):
+    """LW-12 WiFi LED Controller."""
+
+    def __init__(self, name, host, port, rgb_color, brightness, effect,
+                 transition):
+        import lw12
+
+        self._light = lw12.LW12Controller(host, port)
+        self._name = name
+        self._host = host
+        self._port = port
+        self._rgb_color = rgb_color
+        self._brightness = brightness
+        self._state = None
+        self._effect = effect
+        self._transition_speed = transition
+        # Setup feature list
+        self._supported_features = SUPPORT_BRIGHTNESS
+        self._supported_features |= SUPPORT_EFFECT
+        self._supported_features |= SUPPORT_RGB_COLOR
+        self._supported_features |= SUPPORT_TRANSITION
+
+    @property
+    def name(self):
+        """Return the display name of the controlled light."""
+        return self._name
+
+    @property
+    def brightness(self):
+        """Return the brightness of the light."""
+        return self._brightness
+
+    @property
+    def rgb_color(self):
+        """Read back the color of the light.
+
+        Returns [r, g, b] list with values in range of 0-255.
+        """
+        return self._rgb_color
+
+    @property
+    def effect(self):
+        """Return current light effect."""
+        if self._effect is None:
+            return None
+        return self._effect.replace('_', ' ').title()
+
+    @property
+    def is_on(self):
+        """Return true if light is on."""
+        return self._state
+
+    @property
+    def supported_features(self):
+        return self._supported_features
+
+    @property
+    def effect_list(self):
+        """Return a list of available effects.
+
+        Use the Enum element name for display.
+        """
+        import lw12
+        return [effect.name.replace('_', ' ').title()
+                for effect in lw12.LW12_EFFECT]
+
+    @property
+    def assumed_state(self) -> bool:
+        """Return False if unable to access real state of the entity."""
+        return True
+
+    def turn_on(self, **kwargs):
+        """Instruct the light to turn on."""
+        import lw12
+        self._light.light_on()
+        if self._effect is not None:
+            # Effect is set as default: enable effect.
+            kwargs['effect'] = self._effect
+
+        if ATTR_RGB_COLOR in kwargs:
+            rgb = kwargs.get(ATTR_RGB_COLOR)
+            self._rgb_color = rgb
+            self._light.set_color(rgb[0], rgb[1], rgb[2])
+            # Sending UDP messages to quickly after the previous message
+            # the new command is ignored. Adding a short wait time.
+            time.sleep(.25)
+            brightness = int(self._brightness / 255 * 100)
+            self._light.set_light_option(lw12.LW12_LIGHT.BRIGHTNESS,
+                                         brightness)
```

Comment:
I think you would need another `time.sleep(0.25)` after this (and a couple of other places). Also I think halding `kwargs` with both rgb color and brightness can be handled a bit smarter. currently you're first sending the old brightness and then the new brightness from kwargs.

Reply:
Another `time.sleep(.25)` is not needed here, if we call normal `turn_on` without the brightness setting, the light will "forget" the brightness and set the new color. The platform would still list the brightness as defined by either the default setting defined in the configuration or by previous `turn_on` calls with `brightness` as argument.